src/Core/Framework/Adapter/Cache/InvalidatorStorage/MySQLInvalidatorStorage.php

```diff
@@ -0,0 +1,76 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\Adapter\Cache\InvalidatorStorage;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\TransactionIsolationLevel;
+use Shopware\Core\Framework\DataAbstractionLayer\Doctrine\MultiInsertQueryQueue;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Uuid\Uuid;
+
+/**
+ * @codeCoverageIgnore @see \Shopware\Tests\Integration\Core\Framework\Adapter\Cache\InvalidatorStorage\MySQLInvalidatorStorageTest
+ */
+#[Package('core')]
+class MySQLInvalidatorStorage extends AbstractInvalidatorStorage
+{
+    private const TABLE_NAME = 'invalidation_tags';
+
+    private readonly \Closure $debug;
+
+    public function __construct(private readonly Connection $connection, ?\Closure $debug = null)
+    {
+        $this->debug = $debug ?? (fn () => null)(...);
+    }
+
+    public function store(array $tags): void
+    {
+        if (empty($tags)) {
+            return;
+        }
+
+        $insertQueue = new MultiInsertQueryQueue($this->connection, chunkSize: 1000, ignoreErrors: true);
+        $insertQueue->addInserts(
+            self::TABLE_NAME,
+            array_map(
+                fn (string $tag) => ['id' => Uuid::randomBytes(), 'tag' => $tag],
+                array_values($tags)
+            )
+        );
+
+        $this->connection->transactional(fn () => $insertQueue->execute());
+    }
+
+    public function loadAndDelete(): array
+    {
+        // used so that we don't lock the table for inserts
+        $transactionIsolation = $this->connection->getTransactionIsolation();
```

Comment:
just seeing it now, should we use a try...finally block to set transaction level back in case of any error? and then bubble the error back up?

Reply:
yep makes sense 👍